src/processTargets/modifiers/scopeHandlers/CharacterScopeHandler.ts

```diff
@@ -1,21 +1,48 @@
-import { imap } from "itertools";
-import { NestedScopeHandler } from ".";
-import { GRAPHEME_SPLIT_REGEX } from "../../../core/TokenGraphemeSplitter";
-import { Direction } from "../../../typings/targetDescriptor.types";
-import { generateMatchesInRange } from "../../../util/regex";
+import { flatmap, imap } from "itertools";
+import { Position, Range, TextEditor } from "vscode";
+import { getMatcher } from "../../../core/tokenizer";
+import { Direction, ScopeType } from "../../../typings/targetDescriptor.types";
+import { getDocumentRange } from "../../../util/rangeUtils";
+import { generateMatchesInRange, testRegex } from "../../../util/regex";
 import { PlainTarget } from "../../targets";
+import BaseScopeHandler from "./BaseScopeHandler";
+import { expandPosition, expandRange } from "./expandRange";
 import type { TargetScope } from "./scope.types";
+import { ScopeIteratorRequirements } from "./scopeHandler.types";
 
-export default class CharacterScopeHandler extends NestedScopeHandler {
+const EXPANSION_CHARACTERS = 4;
```

Comment:
Do we really need this? Could we not just yield a line at the time?

Reply:
There is range including line break on each line that we probably could use. This would also help with breaking in the middle of a diacritic sends these can't pass lines breaks.

Reply:
Yeah maybe that is simplest; unfortunate because this was a lot of work 😅. Might be useful to retain some of that machinery for more complex regex-based scopes. Tbh might be useful for expansion when doing pairs

I'll have a look at switching this to use line with breaks and migrating the range generator to a utility

Reply:
How important do you think it is that "char" includes newlines?  I tried to impl your solution by just using a nested scope handler with `line` as the search scope, but because the domain of `line` doesn't include the line break, the nested scope handler will skip the line if `position` is at the end of the line, because it thinks there's nothing left in the domain, even though there is the line break

If we just bail on supporting newlines as "char", then we can just use a nested scope handler with search scope of `line`.  I'd be inclined to do this until we have a use case for using "char" to target line

Alternately, we can duplicate the logic from `LineScopeHandler` https://github.com/cursorless-dev/cursorless/blob/c9e06d765cdb74c9d137d07133904711f0955a1f/src/processTargets/modifiers/scopeHandlers/LineScopeHandler.ts#L21-L30 Not too much code, but not sure it's worth it 🤷‍♂️

wdyt?

Reply:
I don't think it's important at all. It might actually be confusing to include line break.